aiosmtpd/controller.py

```diff
@@ -29,11 +32,21 @@ def __init__(self, handler, loop=None, hostname=None, port=8025, *,
         self.ready_timeout = os.getenv(
             'AIOSMTPD_CONTROLLER_TIMEOUT', ready_timeout)
         self.server_kwargs: Dict[str, Any] = server_kwargs or {}
+        if enable_SMTPUTF8 is not None:
+            # Uncomment next lines when we hit 1.3
+            # warn("enable_SMTPUTF8 will be removed in the future. "
+            #      "Please use server_kwargs instead.", DeprecationWarning)
+            self.server_kwargs["enable_SMTPUTF8"] = enable_SMTPUTF8
+        else:
+            # This line emulates previous behavior of defaulting enable_SMTPUTF8 to
+            # True. Which actually kinda conflicts with SMTP class's default, but it's
+            # explained in the documentation.
+            self.server_kwargs.setdefault("enable_SMTPUTF8", True)
 
     def factory(self):
         """Allow subclasses to customize the handler/server creation."""
-        return SMTP(self.handler, enable_SMTPUTF8=self.enable_SMTPUTF8,
-                    **self.server_kwargs)
+        self.smtpd = SMTP(self.handler, **self.server_kwargs)
+        return self.smtpd
```

Comment:
Without this change, `server_kwargs` contains `enable_SMTPUTF8` key, `factory()` will fail. And this failure is _not_ detected with current code.